### DISCUSSION TITLE
How to get SDL for cross compilation?

### DISCUSSION THREAD

**Discussion by mxlgv** [klange/toaruos#233] [Q&A] [answered]
To port one game, I need the SDL library, as well as the headers for it. I think it's to experiment with https://github.com/klange/SDL, but I still haven't figured out how to put it together.

**Comment by klange** [comment, marked as answer]
There is absolutely nothing special about building SDL:

```
./configure --prefix=/usr --host=x86_64-pc-toaru --target=x86_64-pc-toaru --enable-shared
make
make DESTDIR=/path/to/your/toaruos/base install
```

**Comment by DeafMan1983** [comment]
Please be carefully! Because hobbyist has written backend of yutani with SDL 1.2 
Like backend of X11, Wayland, NS or Cocoa of MacOS and User32 Api of Windows 10

You need write whole backend of SDL2 / SDL3.
You can find example in apps/compositor.c and find yutani_window_t or you can open base/user/include/toaru/yutani.h and find yutani_window_t and you will know how do you write backend c files

Example:
TY_Init()
TY_CreateWindow()
TY_CreateWindowFrom()
TY_ShowWindow()
etc ...
TY = Toaru Yutani

And you can add in SDL2 / SDL3 platform if you find X11_CreateWindow then you will know how do you add backend of Yutani's windowing system.
And makefile, you need add whole backend if you use Yutani Windowing System

If you try - but it is very complicated - When you have high experience of C/H with makefile/configure or autogen.sh etc...
